pkg/volmount/getvolumepathnamesforvolumename.go

```diff
@@ -0,0 +1,69 @@
+package volmount
+
+import (
+	"path/filepath"
+	"syscall"
+	"unicode/utf16"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// utf16ToStringArray returns the UTF-8 encoding of the sequence of UTF-16 sequences s,
+// with a terminating NUL removed. The sequences are terminated by an additional NULL.
+func utf16ToStringArray(s []uint16) (result []string) {
+	start := 0
+
+	for i, v := range s {
+		if v != 0 {
+			continue
+		}
+
+		if start == i {
+			// Empty string, that's the end.
+			return
+		}
+
+		result = append(result, string(utf16.Decode(s[start:i])))
+		start = i + 1
+	}
+
+	// The buffer was incomplete, the expected "additional NULL" wasn't present.
+	// Just return the complete values we did decode, assume the caller intended
+	// to do this, e.g. as a best-effort call without resizing the buffer.
+	return
+}
+
+// GetVolumePathNamesForVolumeName returns a list of mount points for the volumePath
+// (in format '\\?\Volume{GUID}).
+// https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw
+func GetVolumePathNamesForVolumeName(volumePath string) ([]string, error) {
```

Comment:
Interesting. I assumed it should match the Win32 API function name. Although I notice now that I linked to the wrong MSDN page here. Probably a copy-and-paste error from `GetVolumeNameForVolumeMountPoint`.

Reply:
(I fixed the link, but will leave the comment unresolved pending further feedback)

Reply:
I think that since we are wrapping the win32 APIs we can have names that make more sense to the caller. We do that for some of the wrapper functions in here: https://github.com/microsoft/go-winio/blob/master/vhd/vhd.go.

Reply:
All the APIs in that file that aren't directly named for the Win32 API they wrap are also adding functionality. Particularly the `XVirtualDisk`/`XVhd` API distinction there caused me to go in the wrong direction while trying to use those APIs in test suite for these APIs. I intended these functions to be similarly-trivial wrappers that just take care of `*uint16` -> `string`, so I could lean on the MSDN documentation.

That said, I can apply this renaming next time I'm on this code, I'm out of time just now. -_-